Add numeronyms functionality

patrickmarabeas opened the pull request:
Implemented numeronym functionality in a executable Go program which returns a slice of numeronyms given an input of n strings

Fixes #419 

Review of colleague's PR #526

brvishnu92 commented on 04_numeronym/patrickmarabeas/main.go:
```diff
@@ -0,0 +1,36 @@
+func numeronyms(vals ...string) []string {
+	result := make([]string, len(vals))
+	i := 0
+	for _, numeronym := range vals {
```
Wouldn't need to declare i separately this way, `for i, numeronym := range vals {`

patrickmarabeas replied:
Good point! Was in the mindset of looping maps : )

brvishnu92 commented on 04_numeronym/patrickmarabeas/main.go:
```diff
@@ -0,0 +1,36 @@
+
+// Convert string to a numeronym
+func generateNumeronym(s string) string {
+	length := len(s)
```
length variable creation may be unnecessary as its used in only one case.

So perhaps `if len(s)<4` would suffice.

patrickmarabeas replied:
It's also used in the return statement twice.

brvishnu92 commented on 04_numeronym/patrickmarabeas/main_test.go:
```diff
@@ -0,0 +1,80 @@
+	got := buf.String()
+
+	t.Run("Main function", func(t *testing.T) {
+		if expected != got {
```
Consider using testify package  to validate test cases `"github.com/stretchr/testify/assert"`. Its stressed by the team. But, will leave this upto you.

patrickmarabeas replied:
I'll look into it for future labs, thanks.

pentaphobe reviewed (changes requested):
Looking really good, I feel it's ready for approval once there's a colleague PR # referenced

patrickmarabeas commented:
@pentaphobe PR # reference added.

patrickmarabeas commented:
Updated doc strings

patrickmarabeas committed:
Add numeronyms functionality

Implemented numeronym functionality in a executable Go program which
returns a slice of numeronyms given an input of n strings

Closes Lab 04 (#419)

patrickmarabeas commented:
Updated to handle multi byte characters

pentaphobe commented:
LGTM

@brvishnu92 please confirm you're happy with changes, some weren't dismissed automatically since the code lines weren't changed (see final review point)

brvishnu92 commented:
@patrickmarabeas  LGTM :-)
